app/src/main/java/org/wikipedia/search/SearchFragment.kt

```diff
@@ -0,0 +1,408 @@
+import org.wikipedia.Constants.InvokeSource
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.analytics.IntentFunnel
+import org.wikipedia.analytics.SearchFunnel
+import org.wikipedia.database.contract.SearchHistoryContract
+import org.wikipedia.databinding.FragmentSearchBinding
+import org.wikipedia.history.HistoryEntry
+import org.wikipedia.page.ExclusiveBottomSheetPresenter
+import org.wikipedia.page.PageActivity
+import org.wikipedia.page.PageTitle
+import org.wikipedia.readinglist.AddToReadingListDialog
+import org.wikipedia.readinglist.MoveToReadingListDialog
+import org.wikipedia.readinglist.ReadingListBehaviorsUtil.AddToDefaultListCallback
+import org.wikipedia.readinglist.ReadingListBehaviorsUtil.addToDefaultList
+import org.wikipedia.settings.Prefs
+import org.wikipedia.settings.languages.WikipediaLanguagesActivity
+import org.wikipedia.settings.languages.WikipediaLanguagesFragment
+import org.wikipedia.util.DeviceUtil.hideSoftKeyboard
+import org.wikipedia.util.DeviceUtil.setWindowSoftInputModeResizable
+import org.wikipedia.util.FeedbackUtil.setButtonLongPressToast
+import org.wikipedia.util.FeedbackUtil.showTooltip
+import org.wikipedia.util.ResourceUtil.getThemedColor
+import org.wikipedia.util.StringUtil.listToJsonArrayString
+import org.wikipedia.views.LanguageScrollView
+import org.wikipedia.views.ViewUtil.formatLangButton
+import java.util.*
+
+class SearchFragment : Fragment(), SearchResultsFragment.Callback, RecentSearchesFragment.Callback, LanguageScrollView.Callback {
+    private var _binding: FragmentSearchBinding? = null
+    private val binding get() = _binding!!
+    private val disposables = CompositeDisposable()
+    private var app: WikipediaApp? = null
+    private var funnel: SearchFunnel? = null
+    private var invokeSource: InvokeSource? = null
+    var searchLanguageCode: String? = null
+        private set
+    private var tempLangCodeHolder: String? = null
+    private var langBtnClicked = false
+    private var initialLanguageList: String? = null
+
+    private var isSearchActive = false
+
+    private var query: String? = null
+    private val bottomSheetPresenter = ExclusiveBottomSheetPresenter()
+    private var recentSearchesFragment: RecentSearchesFragment? = null
+    private var searchResultsFragment: SearchResultsFragment? = null
+
+    private val searchCloseListener = SearchView.OnCloseListener {
+        closeSearch()
+        funnel!!.searchCancel(searchLanguageCode)
+        false
+    }
+
+    private val searchQueryListener: SearchView.OnQueryTextListener = object : SearchView.OnQueryTextListener {
+        override fun onQueryTextSubmit(queryText: String): Boolean {
+            hideSoftKeyboard(requireActivity())
+            return true
+        }
+
+        override fun onQueryTextChange(queryText: String): Boolean {
+            binding.searchCabView.setCloseButtonVisibility(queryText)
+            startSearch(queryText.trim { it <= ' ' }, false)
```

Comment:
What's the benefit of this trim syntax? (why not just use the plain `trim()` function?)